build/cli/src/lib.rs

```diff
@@ -413,11 +414,15 @@ impl Processor {
                 let mut config = enso_build::engine::BuildConfigurationFlags::default();
                 for arg in which {
                     match arg {
-                        Tests::Jvm => config.test_jvm = true,
-                        Tests::StandardLibrary => config.test_standard_library = true,
+                        Tests::Jvm => {
+                            config.test_jvm = true;
+                            // We also test the Java parser integration when running the JVM tests.
+                            config.test_java_generated_from_rust = true;
+                        },
+                        Tests::StandardLibrary => config.add_standard_library_test_selection(StandardLibraryTestsSelection::All),
+                        Tests::StdSnowflake => config.add_standard_library_test_selection(StandardLibraryTestsSelection::Selected(vec!["Snowflake_Tests".to_string()])),
                     }
                 }
-                config.test_java_generated_from_rust = true;
```

Comment:
It seems that parser tests were being always run. They are rather quick but that seemed wrong - if I request `stdlib` (or now `std-snowflake`) tests, I don't want anything else.